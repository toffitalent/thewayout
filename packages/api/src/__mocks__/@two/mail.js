const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
