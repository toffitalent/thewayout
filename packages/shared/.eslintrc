{
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": ["tsconfig.json", "packages/shared/tsconfig.json"]
      }
    }
  },
  "rules": {
    "no-param-reassign": "off",
    "react/react-in-jsx-scope": "off"
  }
}
