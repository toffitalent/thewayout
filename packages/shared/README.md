# @two/shared

Copyright (c) 2023-Present The Way Out, Inc. All Rights Reserved.

## Development

This package contains code shared between the [api](../api/README.md), [app](../app/README.md), and [web](../web/README.md) packages and is not intended to run in a standalone manner. See the READMEs of those packages for information on starting a development environment.

## Testing

To run all component and unit tests and get coverage reports, simply run `yarn workspace @dlatwobs/shared test`.
