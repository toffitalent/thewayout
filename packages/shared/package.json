{
  "name": "@two/shared",
  "version": "1.0.0",
  "author": "Disruptive Labs",
  "license": "UNLICENSED",
  "private": true,
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "scripts": {
    "build": "tsc -b tsconfig.build.json",
    "clean": "rimraf build coverage",
    "test": "jest --coverage",
    "type-check": "tsc"
  },
  "dependencies": {
    "@disruptive-labs/ui": "file:../../vendor/@disruptive-labs/ui",
    "axios": "^0.27.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/dom": "^9.3.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.2",
    "@types/react": "^18.2.14",
    "babel-jest": "^29.5.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "jest-junit": "^16.0.0",
    "react-test-renderer": "^18.2.0",
    "rimraf": "^3.0.2",
    "sass": "^1.63.6",
    "typescript": "^5.2.2"
  },
  "jest": {
    "clearMocks": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.{ts,tsx}"
    ],
    "globalSetup": "<rootDir>/test/setup.js",
    "moduleNameMapper": {
      "^.+\\.module\\.(css|sass|scss)$": "identity-obj-proxy",
      "^.+\\.(css|sass|scss)$": "<rootDir>/test/styleMock.js",
      "^.+\\.(jpg|jpeg|png|gif|webp|svg)$": "<rootDir>/test/fileMock.js",
      "^@test$": "<rootDir>/test/index.tsx"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/test/jest.test.ts"
    ],
    "testEnvironment": "jsdom",
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/test/"
    ],
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": [
        "babel-jest",
        {
          "presets": [
            "next/babel"
          ]
        }
      ]
    },
    "transformIgnorePatterns": [
      "node_modules/(?!(@?disruptive-labs.*|@popperjs/core|es-cookie)/)"
    ]
  },
  "jest-junit": {
    "outputDirectory": "./artifacts/jest",
    "outputName": "results.xml"
  }
}
