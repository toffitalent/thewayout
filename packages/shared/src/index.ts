// NOTE: React components should not exported here

export * from './constants';
export * from './types';
export * from './utils';
