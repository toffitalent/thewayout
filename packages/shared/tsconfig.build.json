{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "incremental": true,
    "module": "commonjs",
    "noEmit": false,
    "outDir": "./build",
    "rootDir": "./src",
    "tsBuildInfoFile": "./build/.tsbuildinfo"
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["build", "node_modules", "**/*.test.ts", "**/*.test.tsx", "test"]
}
