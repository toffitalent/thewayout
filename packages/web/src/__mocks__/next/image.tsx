import * as React from 'react';

// eslint-disable-next-line react/jsx-no-useless-fragment
const mock = (props: { children: React.ReactElement }): React.ReactElement => <>{props.children}</>;

export default mock;
