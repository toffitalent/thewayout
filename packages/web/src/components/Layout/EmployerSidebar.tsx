import {
  Avatar,
  Flex,
  Menu,
  MenuGroup,
  MenuItem,
  MenuList,
  MenuTrigger,
} from '@disruptive-labs/ui';
import BriefcaseIcon from '@disruptive-labs/ui/dist/icons/Briefcase';
import DotsVerticalIcon from '@disruptive-labs/ui/dist/icons/DotsVertical';
import { Sidebar, SidebarItem, SidebarSection } from '@disruptive-labs/ui/dist/layouts';
import Image from 'next/legacy/image';
import Link from 'next/link';
import { useRouter } from 'next/router';
import logoImg from '@app/assets/images/logo.png';
import { logout, selectAuthUser } from '@app/features/auth';
import { useAppDispatch, useAppSelector } from '@app/hooks';
import styles from './RspSidebar.module.scss';

export const EmployerSidebar = () => {
  const router = useRouter();
  const user = useAppSelector(selectAuthUser);
  const dispatch = useAppDispatch();

  if (!user || !user.employer) {
    return null;
  }
  const { employer } = user;

  return (
    <Sidebar
      className={styles.sidebar}
      logo={
        <Flex alignItems="center" display="flex">
          <Link href="/" passHref className={styles.logo}>
            <Image src={logoImg} alt="" width={0} height={0} />
          </Link>
        </Flex>
      }
    >
      <SidebarSection>
        <SidebarItem
          accessoryLeft={
            <Avatar name={employer.name} className={styles.avatar} src={employer.avatar} />
          }
        >
          {employer.name}
        </SidebarItem>
      </SidebarSection>

      <SidebarSection fontSize="lg">
        <SidebarItem
          active={router.asPath === '/employer/'}
          accessoryLeft={<BriefcaseIcon />}
          wrapper="a"
          wrapperProps={{ href: '/employer' }}
        >
          Job Postings
        </SidebarItem>
      </SidebarSection>

      <SidebarSection separated>
        <SidebarItem
          accessoryLeft={
            <Avatar size="sm" name={`${user.firstName} ${user.lastName}`} src={user.avatar} />
          }
          accessoryRight={
            <Menu>
              <MenuTrigger>
                <DotsVerticalIcon height={6} width={6} cursor="pointer" />
              </MenuTrigger>
              <MenuList>
                <MenuGroup>
                  <MenuItem minWidth={40} onClick={() => router.push('/employer/account-settings')}>
                    Account settings
                  </MenuItem>
                  <MenuItem minWidth={40} onClick={() => dispatch(logout())}>
                    Logout
                  </MenuItem>
                </MenuGroup>
              </MenuList>
            </Menu>
          }
        >{`${user.firstName} ${user.lastName}`}</SidebarItem>
      </SidebarSection>
    </Sidebar>
  );
};
