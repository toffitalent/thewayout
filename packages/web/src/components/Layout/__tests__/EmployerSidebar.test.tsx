import { useRouter } from 'next/router';
import { RspRole, UserType } from '@two/shared';
import { mockState, render, waitFor } from '@test';
import { EmployerSidebar } from '../EmployerSidebar';

describe('EmployerSidebar', () => {
  beforeEach(() => {
    (useRouter as jest.Mock).mockImplementation(() => ({
      asPath: '/',
    }));
  });

  test('renders correctly', async () => {
    const { container, findByText } = render(<EmployerSidebar />, {
      initialState: {
        ...mockState,
        auth: {
          user: { employer: { name: 'Test' }, type: UserType.Employer },
        },
      },
    });

    await waitFor(async () => expect(await findByText('Job Postings')).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });
});
