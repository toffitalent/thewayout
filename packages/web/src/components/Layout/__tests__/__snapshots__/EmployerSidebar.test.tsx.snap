// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployerSidebar renders correctly 1`] = `
<div>
  <div
    class="sidebar grey sidebar"
  >
    <div
      class="navbar"
    >
      <button
        aria-haspopup="dialog"
        aria-label="Open sidebar navigation"
        class="button sm clear c-grey iconButton sm round fontSize-2xl"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon"
          focusable="false"
          stroke-width="2"
          viewBox="0 0 24 24"
        >
          <path
            d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
          />
        </svg>
      </button>
      <div
        class="logo"
      >
        <div
          class="alignItems-center display-flex"
        >
          <a
            class="logo"
            href="/"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%270%27%20height=%270%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt=""
                data-nimg="intrinsic"
                decoding="async"
                src="/_next/image?url=%2Ftest-file-stub&w=16&q=75"
                srcset="/_next/image?url=%2Ftest-file-stub&w=16&q=75 1x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </div>
      </div>
    </div>
    <nav
      class="sidebarNav"
    >
      <div
        class="logo"
      >
        <div
          class="alignItems-center display-flex"
        >
          <a
            class="logo"
            href="/"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%270%27%20height=%270%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt=""
                data-nimg="intrinsic"
                decoding="async"
                src="/_next/image?url=%2Ftest-file-stub&w=16&q=75"
                srcset="/_next/image?url=%2Ftest-file-stub&w=16&q=75 1x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </div>
      </div>
      <div
        class="section"
      >
        <ul
          class="sectionList"
        >
          <li
            class="item"
          >
            <span
              class="itemWrapper"
            >
              <span
                class="itemAccessory"
              >
                <div
                  class="avatar avatar"
                >
                  <div
                    aria-label="Test"
                    class="initials"
                  >
                    T
                  </div>
                </div>
              </span>
              Test
            </span>
          </li>
        </ul>
      </div>
      <div
        class="section fontSize-lg"
      >
        <ul
          class="sectionList"
        >
          <li
            class="item"
          >
            <a
              class="itemWrapper"
              href="/employer"
            >
              <span
                class="itemAccessory"
              >
                <svg
                  aria-hidden="true"
                  class="icon"
                  focusable="false"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    stroke="none"
                  />
                  <path
                    d="M3 9a2 2 0 012-2h14a2 2 0 012 2v9a2 2 0 01-2 2H5a2 2 0 01-2-2zM8 7V5a2 2 0 012-2h4a2 2 0 012 2v2M12 12v.01M3 13a20 20 0 0018 0"
                  />
                </svg>
              </span>
              Job Postings
            </a>
          </li>
        </ul>
      </div>
      <div
        class="section sectionSeparated"
      >
        <ul
          class="sectionList"
        >
          <li
            class="item"
          >
            <span
              class="itemWrapper"
            >
              <span
                class="itemAccessory"
              >
                <div
                  class="avatar s-sm"
                >
                  <div
                    aria-label="undefined undefined"
                    class="initials"
                  >
                    uu
                  </div>
                </div>
              </span>
              undefined undefined
              <span
                class="itemAccessory itemAccessoryRight"
              >
                <svg
                  aria-controls="menu-:r1:"
                  aria-expanded="false"
                  aria-haspopup="menu"
                  aria-hidden="true"
                  class="icon width-6 height-6"
                  cursor="pointer"
                  focusable="false"
                  id="menu-:r1:-trigger"
                  role="button"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    stroke="none"
                  />
                  <path
                    d="M11 12a1 1 0 102 0 1 1 0 10-2 0M11 19a1 1 0 102 0 1 1 0 10-2 0M11 5a1 1 0 102 0 1 1 0 10-2 0"
                  />
                </svg>
                <div
                  class="menu"
                  data-popper-escaped="true"
                  data-popper-placement="bottom-start"
                  data-popper-reference-hidden="true"
                  style="visibility: hidden; position: absolute; top: 0px; left: 0px; min-width: max-content; will-change: transform, visibility; transform: translate(0px, 0px);"
                >
                  <div
                    aria-orientation="vertical"
                    class="list"
                    id="menu-:r1:"
                    role="menu"
                    tabindex="-1"
                  >
                    <div
                      class="group"
                      role="group"
                    >
                      <button
                        class="button md text c-primary item minWidth-40"
                        role="menuitem"
                        tabindex="-1"
                        type="button"
                      >
                        Account settings
                      </button>
                      <button
                        class="button md text c-primary item minWidth-40"
                        role="menuitem"
                        tabindex="-1"
                        type="button"
                      >
                        Logout
                      </button>
                    </div>
                  </div>
                </div>
              </span>
            </span>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
